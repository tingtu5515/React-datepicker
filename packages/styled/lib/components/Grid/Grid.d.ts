import {
  GridAutoColumnsProps,
  GridAutoFlowProps,
  GridAutoRowsProps,
  AlignItemsProps,
  JustifyContentProps,
  GridColumnGapProps,
  GridGapProps,
  GridRowGapProps,
  GridTemplatesAreasProps,
  GridTemplatesColumnsProps,
  GridTemplatesRowsProps,
  SpaceProps,
} from 'styled-system'
interface GridProps
  extends GridAutoColumnsProps,
    GridAutoFlowProps,
    GridAutoRowsProps,
    AlignItemsProps,
    JustifyContentProps,
    GridColumnGapProps,
    GridGapProps,
    GridRowGapProps,
    GridTemplatesAreasProps,
    GridTemplatesColumnsProps,
    SpaceProps,
    GridTemplatesRowsProps {
  daySizeGridTemplateColumns?: number | (number | null)[] | undefined
}
declare const Grid: import('styled-components').StyledComponent<'div', any, GridProps, never>
export default Grid
