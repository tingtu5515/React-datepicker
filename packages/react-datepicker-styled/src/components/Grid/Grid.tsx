import styled from 'styled-components'
import {
  gridAutoColumns,
  GridAutoColumnsProps,
  gridAutoFlow,
  GridAutoFlowProps,
  gridAutoRows,
  GridAutoRowsProps,
  alignItems,
  AlignItemsProps,
  justifyContent,
  JustifyContentProps,
  gridColumnGap,
  GridColumnGapProps,
  gridGap,
  GridGapProps,
  gridRowGap,
  GridRowGapProps,
  gridTemplateAreas,
  GridTemplatesAreasProps,
  gridTemplateColumns,
  GridTemplatesColumnsProps,
  gridTemplateRows,
  GridTemplatesRowsProps,
  space,
  SpaceProps,
} from 'styled-system'

interface GridProps
  extends GridAutoColumnsProps,
    GridAutoFlowProps,
    GridAutoRowsProps,
    AlignItemsProps,
    JustifyContentProps,
    GridColumnGapProps,
    GridGapProps,
    GridRowGapProps,
    GridTemplatesAreasProps,
    GridTemplatesColumnsProps,
    SpaceProps,
    GridTemplatesRowsProps {}

const Grid = styled('div')<GridProps>`
  display: grid;
  ${gridAutoColumns}
  ${gridAutoFlow}
  ${gridAutoRows}
  ${gridColumnGap}
  ${gridGap}
  ${gridRowGap}
  ${gridTemplateAreas}
  ${gridTemplateColumns}
  ${gridTemplateRows}
  ${alignItems}
  ${justifyContent}
  ${space}
`

export default Grid
