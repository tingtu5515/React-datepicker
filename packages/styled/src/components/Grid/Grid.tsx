import styled from 'styled-components'
import {
  gridAutoColumns,
  GridAutoColumnsProps,
  gridAutoFlow,
  GridAutoFlowProps,
  gridAutoRows,
  GridAutoRowsProps,
  alignItems,
  AlignItemsProps,
  justifyContent,
  JustifyContentProps,
  gridColumnGap,
  GridColumnGapProps,
  gridGap,
  GridGapProps,
  gridRowGap,
  GridRowGapProps,
  gridTemplateAreas,
  GridTemplatesAreasProps,
  gridTemplateColumns,
  GridTemplateColumnsProps,
  gridTemplateRows,
  GridTemplateRowsProps,
  space,
  SpaceProps,
  style,
  compose,
} from 'styled-system'

interface GridProps
  extends GridAutoColumnsProps,
    GridAutoFlowProps,
    GridAutoRowsProps,
    AlignItemsProps,
    JustifyContentProps,
    GridColumnGapProps,
    GridGapProps,
    GridRowGapProps,
    GridTemplatesAreasProps,
    GridTemplateColumnsProps,
    SpaceProps,
    GridTemplateRowsProps {
  daySizeGridTemplateColumns?: number | (number | null)[] | undefined
  numberOfMonthsGridTemplateColumns?: number | (number | null)[] | undefined
}

const daySizeGridTemplateColumns = style({
  // React prop name and CSS property
  prop: 'daySizeGridTemplateColumns',
  // CSS property (if different from prop argument)
  cssProperty: 'gridTemplateColumns',
  // key for theme values
  key: 'gridTemplateColumns',
  // accessor function for transforming the value
  transformValue: n => `repeat(7, ${n}px)`,
  // add a fallback scale object or array, if theme is not present
  scale: [0, 4, 8, 16, 32],
})

const composeGridStyles = compose(
  gridAutoColumns,
  gridAutoFlow,
  gridAutoRows,
  gridColumnGap,
  gridGap,
  gridRowGap,
  gridTemplateAreas,
  gridTemplateColumns,
  gridTemplateRows,
  alignItems,
  justifyContent,
  space,
)

const Grid = styled('div')<GridProps>`
  display: grid;
  ${composeGridStyles}
  ${daySizeGridTemplateColumns}
`

export default Grid
