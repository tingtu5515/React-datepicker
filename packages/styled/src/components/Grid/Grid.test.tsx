import React from 'react'
import {render} from 'react-testing-library'
import Grid from '.'

test('should match snapshot', () => {
  const {container} = render(
    <Grid
      gridTemplateColumns="repeat(6, 1fr)"
      gridTemplateRows="repeat(6, 1fr)"
      gridTemplateAreas={'"c1 c2 c3" "c4 c5 c6"'}
      gridGap="10px 10px"
      gridColumnGap="10px"
      gridRowGap="10px"
      justifyContent="initial"
      alignItems="initial"
    >
      <div>test1</div>
      <div>test2</div>
      <div>test3</div>
      <div>test4</div>
      <div>test5</div>
    </Grid>,
  )
  expect(container).toMatchSnapshot()
})
