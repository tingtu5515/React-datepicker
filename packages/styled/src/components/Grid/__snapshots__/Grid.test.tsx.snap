// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-column-gap: 10px;
  grid-gap: 10px 10px;
  grid-row-gap: 10px;
  grid-template-areas: "c1 c2 c3" "c4 c5 c6";
  grid-template-columns: repeat(6,1fr);
  grid-template-rows: repeat(6,1fr);
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
}

<div>
  <div
    class="c0"
  >
    <div>
      test1
    </div>
    <div>
      test2
    </div>
    <div>
      test3
    </div>
    <div>
      test4
    </div>
    <div>
      test5
    </div>
  </div>
</div>
`;
